File: src/main/groovy/org/featx/spec/constant/ErrorCode.java
package org.featx.spec.constant;

/**
 * @author Excepts
 * @since 2019/10/27 11:48
 */
public class ErrorCode {

    public static final Integer NO_ERROR = 1511200;
    /**
     * Lost Parameter
     */
    public static final Integer PARAMETER_LOST = 4000010;
    /**
     * Parameter type error
     */
    public static final Integer PARAMETER_TYPE = 4000011;
    /**
     * Parameter valid error
     */
    public static final Integer PARAMETER_INVALID = 4000012;
    /**
     * The api method required authorized
     */
    public static final Integer UNAUTHORIZED = 4010000;
    /**
     * The authorization credential error. e.g. username, and password not matched
     */
    public static final Integer UNAUTHORIZED_CREDENTIAL = 4010001;
    /**
     * The authorization access token expired
     */
    public static final Integer UNAUTHORIZED_TOKEN_EXPIRED = 4010002;
    /**
     * Not permit to access the resource
     */
    public static final Integer PERMISSION_DENIED = 4030000;
    /**
     * Not found
     */
    public static final Integer NOT_FOUND = 4040000;
    /**
     * Duplicated key, resource id existed
     */
    public static final Integer DUPLICATED_KEY = 4090000;
    /**
     * SERVER_ERROR
     */
    public static final Integer SERVER_ERROR = 5000000;
    /**
     * RPC error.
     */
    public static final Integer SERVER_REMOTE_PROCEDURE_CALL = 5000001;
}

File: src/main/kotlin/org/featx/jvmon/exception/ErrorCode.kt
package org.featx.jvmon.exception

/**
 * @author Excepts
 * @since 2019/10/12 11:48
 */
object ErrorCode {

    const val NO_ERROR = 0
    /**
     * 缺少参数
     */
    const val PARAMETER_LOST = 4000010
    /**
     * 参数类型错误
     */
    const val PARAMETER_TYPE = 4000011
    /**
     * 参数无效，更细的业务校验不通过
     */
    const val PARAMETER_INVALID = 4000012
    /**
     * 查找不到相关信息
     */
    const val NOT_FOUND = 4040000
}
